ui/app/components/secret-engine/configure-azure.ts

```diff
@@ -0,0 +1,181 @@
+    const wifAttributesSet = !!identityTokenAudience || !!identityTokenTtl;
+    const azureAttributesSet = !!clientSecret || !!rootPasswordTtl;
+    // if any WIF attributes have been set in the model, set accessType to 'wif'
+    this.accessType = wifAttributesSet ? 'wif' : 'azure';
+    // if there are either WIF or azure attributes, disable user's ability to change accessType
+    this.disableAccessType = wifAttributesSet || azureAttributesSet;
+  }
+
+  @action continueSubmitForm() {
+    this.saveIssuerWarning = '';
+    this.save.perform();
+  }
+
+  // check if the issuer has been changed to show issuer modal
+  // continue saving the configuration
+  submitForm = task(
+    waitFor(async (event: Event) => {
+      event?.preventDefault();
+      this.resetErrors();
+      const { issuerConfig } = this.args;
+
+      if (issuerConfig?.hasDirtyAttributes) {
+        // if the issuer has changed show modal with warning that the config will change
+        // if the modal is shown, the user has to click confirm to continue saving
+        this.saveIssuerWarning = `You are updating the global issuer config. This will overwrite Vault's current issuer ${
+          issuerConfig.queryIssuerError ? 'if it exists ' : ''
+        }and may affect other configurations using this value. Continue?`;
+        // exit task until user confirms
+        return;
+      }
+      await this.save.perform();
+    })
+  );
+
+  save = task(
+    waitFor(async () => {
+      const { model, issuerConfig } = this.args;
+      // check if any of the model attributes have changed
+      // if no changes, transition and notify user
+      // if changes, save the model and notify user
+      // note: "backend" dirties model state so explicity ignore it here
+      const attrChanged = Object.keys(model?.changedAttributes()).some((item) => item !== 'backend');
+      const issuerAttrChanged = issuerConfig?.hasDirtyAttributes;
+      if (!attrChanged && !issuerAttrChanged) {
+        this.flashMessages.info('No changes detected.');
+        this.transition();
+        return;
+      }
+
+      const modelSaved = attrChanged ? await this.saveModel() : false;
+      const issuerSaved = issuerAttrChanged ? await this.updateIssuer() : false;
+
+      if (modelSaved || issuerSaved) {
+        this.transition();
+      } else {
+        // otherwise there was a failure and we should not transition and exit the function
+        return;
+      }
+    })
+  );
+
+  async updateIssuer(): Promise<boolean> {
+    try {
+      await this.args.issuerConfig.save();
+      this.flashMessages.success('Issuer saved successfully');
+      return true;
+    } catch (e) {
+      this.flashMessages.danger(`Issuer was not saved: ${errorMessage(e, 'Check Vault logs for details.')}`);
+      return false;
+    }
+  }
+
+  async saveModel(): Promise<boolean> {
+    const { backendPath, model } = this.args;
+    try {
+      await model.save();
```

Comment:
since the `clientSecret` isn't returned by the API...does saving it completely wipe it out? Wondering if we'll need to perform a patch here on edit instead..